src/pokedex_data.c

```diff
@@ -0,0 +1,1240 @@
+#include "pokedex_data.h"
+
+#include <nitro.h>
+#include <string.h>
+
+#include "constants/species.h"
+
+#include "heap.h"
+#include "inlines.h"
+#include "pokemon.h"
+#include "savedata.h"
+#include "unk_020986CC.h"
+
+#define DEX_SIZE_U32           ((int)((NATIONAL_DEX_COUNT - 1) / 32) + 1) // default 16
+#define MAGIC_NUMBER           3203386110
+#define NUM_MYTHICALS_NATIONAL 11
+#define NATIONAL_DEX_GOAL      (NATIONAL_DEX_COUNT - NUM_MYTHICALS_NATIONAL)
+#define UNOWN_COUNT            28
+#define DEOXYS_COUNT           4
+#define ROTOM_COUNT            6
+#define TERMINAL_4BITS         0x7
+#define TERMINAL_BYTE          0xf
+#define TERMINAL_U8            0xff
+#define TERMINAL_U32           0xffffffff
+
+typedef struct PokedexData {
+    u32 magic;
+    u32 caughtPokemon[DEX_SIZE_U32];
+    u32 seenPokemon[DEX_SIZE_U32];
+    u32 recordedGenders[2][DEX_SIZE_U32];
+    u32 spindaForms;
+    u8 shellosForms;
+    u8 gastrodonForms;
+    u8 burmyForms;
+    u8 wormadamForms;
+    u8 unownForms[UNOWN_COUNT];
```

Comment:
Do these keep track of the forms being seen or caught?

Reply:
I think it's forms seen